src/Features/CSharp/Portable/IntroduceVariable/CSharpIntroduceVariableService_IntroduceLocal.cs

```diff
@@ -138,7 +138,7 @@ private LambdaExpressionSyntax GetParentLambda(ExpressionSyntax expression, ISet
                     return (LambdaExpressionSyntax)current.Parent;
                 }
 
-                current = current.Parent as ExpressionSyntax;
+                current = current.Parent?.FirstAncestorOrSelf<ExpressionSyntax>();
```

Comment:
i don't know why this can't just be `current = current.parent`.  Or, the entire loop could be replaced with `expression.Ancestors().FirstOrDefault(lambdas.Contains)`

Reply:
Yes, that would be even better! Interestingly, though, simply replacing
```cs
var parentLambda = GetParentLambda(expression, lambdas);
```
on line 85 with
```cs
var parentLambda = expression.Ancestors().FirstOrDefault(lambdas.Contains);
```
breaks 4 of the IntroduceVariableTests. I'll look into it.

Reply:
@CyrusNajmabadi I'm running into the cast on line 110 in the method called `IntroduceLocalDeclarationIntoLambda`:
```cs
var oldBody = (ExpressionSyntax)oldLambda.Body;
```
In one of the 4 tests that break, however, the body of the lambda in question is not an expression but a block.

I'm afraid this could get out of hand quickly, refactoring-wise. Would you like me to continue, or will you accept my one-line first contribution as it is now? :)

Reply:
@CyrusNajmabadi That doesn't work, unfortunately! There is test called `TestIntroduceLocalInExpressionBodiedMethodWithExpressionBodiedParenthesizedLambdaExpression` that has an initial situation looking like
```cs
using System;
class TestClass
{
    Func<int, int> Y() => (f) => f * 9; //<== this '9' being the expression for which to create a local
}
```
and which expects the local to be generated in the body of the arrow expression `=> (f) => f * 9` and _not_ in the lambda `(f) => f * 9`, which we would have found using your suggestion.

Reply:
@CyrusNajmabadi What I could do, however, is to remove `GetParentLambda`, replace the call to it in `GetContainerToGenerateInto` with
```cs
var parentLambda = expression.Ancestors().FirstOrDefault(lambdas.Contains);
```
(like you suggested) but make sure that `lambdas` contains only expression-bodied ones by adding a `Where(...` to the initialization of `lambdas`. Does that sound good?

Reply:
I would still just ahve a single loop that just returns the first viable container (either a block, `=> body` or `lamdba`)

Reply:
@CyrusNajmabadi Done!

Reply:
I really like what this code ended up with, not only simpler/more performant, but also provides better results 👍